ui/app/AppLayouts/Onboarding/views/ConfirmPasswordView.qml

```diff
@@ -0,0 +1,136 @@
+                color: Theme.palette.baseColor1
+            }
+
+            StatusBaseText {
+                anchors.horizontalCenter: parent.horizontalCenter
+                text: qsTr("If you loose your password you will loose access to  your Status profile.")
+                font.pixelSize: 12
+                color: Theme.palette.baseColor1
+            }
+        }
+
+        // TODO replace with StatusInput as soon as it supports password
+        Input {
+            id: confPswInput
+            width: parent.width
+            enabled: !submitBtn.loading
+            placeholderText: submitBtn.loading ? qsTr("Connecting...") : qsTr("Confirm you password (again)")
+            textField.echoMode: TextInput.Password
+            textField.validator: RegExpValidator { regExp: /^[!-~]+$/ } // That incudes NOT extended ASCII printable characters less space
+            keepHeight: true
+            textField.rightPadding: showHideCurrentIcon.width + showHideCurrentIcon.anchors.rightMargin + Style.current.padding / 2
+
+            StatusFlatRoundButton {
+                id: showHideCurrentIcon
+                visible: confPswInput.text !== ""
+                anchors.verticalCenter: parent.verticalCenter
+                anchors.right: parent.right
+                anchors.rightMargin: 16
+                width: 24
+                height: 24
+                icon.name: confPswInput.textField.echoMode === TextInput.Password ? "show" : "hide"
+                icon.color: Theme.palette.baseColor1
+
+                onClicked: confPswInput.textField.echoMode === TextInput.Password ? confPswInput.textField.echoMode = TextInput.Normal : confPswInput.textField.echoMode = TextInput.Password
+            }
+        }
+
+        // Just a column filler to fit the design
+        Item {
+            height: Style.current.padding
+            width: parent.width
+        }
+
+        StatusButton {
+            id: submitBtn
+            anchors.horizontalCenter: parent.horizontalCenter
+            text: qsTr("Finalize Status Password Creation")
+            enabled: !submitBtn.loading && confPswInput.text === root.password
+
+            property Timer sim: Timer {
+                id: pause
+                interval: 20
+                onTriggered: {
+                    // Create new password call action to the backend
+                    OnboardingStore.onBoardingModul.storeSelectedAccountAndLogin(root.password)
+                    applicationWindow.prepareForStoring(root.password, false)
+                }
+            }
+
+            onClicked: {
+                confPswInput.text = ""
+                submitBtn.loading = true
+                //create password operation blocks the UI so loading = true; will never
+                //have any affect until changePassword/createPassword is done. Getting around it with a
+                //small pause (timer) in order to get the desired behavior
+                pause.start()
```

Comment:
Would `Qt.callLater` fix the issue?

Reply:
I've tried with your suggestion and the feeling for the user is that when clicked, the application is like blocked for 1/2 seconds before deactivating the button and showing the text "Connecting...". In case of using the timer, the button is immediately deactivated and so on... Well... the difference is only a second but I get a better feeling with the timer option. I don't know if there is another more elegant way to approach that...